wpimath/src/main/java/edu/wpi/first/math/interpolation/TimeInterpolatableBuffer.java

```diff
@@ -0,0 +1,139 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.math.interpolation;
+
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+import edu.wpi.first.wpiutil.math.MathUtil;
+
+/**
+ * The TimeInterpolatableBuffer provides an easy way to estimate past measurements. One
+ * application might be in conjunction with the DifferentialDrivePoseEstimator,
```

Comment:
Why would this be useful with a DifferentialDrivePoseEstimator? The estimator itself takes in vision measurements along with timestamps and stores them in its own internal buffer, eliminating the need for users to store their measurements in this class.

Reply:
If you have a measurement of target pose relative to the robot at some time `t`, you might also need the pose of the robot at that time `t` as well to get a global measurement.